src/services/auth.ts

```diff
@@ -0,0 +1,21 @@
+import {Product} from "../schemas/product.entity";
+import {User} from "../schemas/user.entity";
+
+export async function regUserService(email, password) {
+
+
+  const existingUser = await User.findOne({ email });
+  if (existingUser) {
+    return { status: 409, message:'User with this email already exists.'};
```

Comment:
That's persistence layer according to its function, not service. Nevertheless, it shouldn't know anything about statuses, that's what controller should do. Statuses are never set below controller level. But you can throw a custom error here, e.g. UserAlreadyExistsError, and set a statusCode and default message for it.